test/integration/features/profile.feature

```diff
@@ -2,104 +2,211 @@
+    And profile "minishift" should be an active profile
+     When executing "minishift profile set <profilename>" fails
+     Then stderr should contain
+     """
+     A profile name must be provided. Run `minishift profile list` for a list of existing profiles.
+     """
 
-Scenario: User should be able to delete profile 'foo'
-    When executing "minishift profile delete foo --force" succeeds
-    Then stdout should contain
-    """
-    Profile 'foo' deleted successfully
-    """
+  Examples: Empty profile name
+    | profilename |
+    | ''          |
+    | ' '         |
+    
+  Scenario Outline: As user, I cannot create profile with special character in profile name
+    Given profile "minishift" has state "Does Not Exist"
+    And profile "minishift" should be an active profile
+     When executing "minishift profile set <profilename>" fails
+     Then stderr should contain
+     """
+     Profile names must consist of alphanumeric characters only.
+     """
 
-Scenario: User can not delete default profile 'minishift'
-    When executing "minishift profile delete minishift --force"
-    Then exitcode should equal "1"
-     And stderr should contain
+  Examples: Wrong profile names
+    | profilename |
+    | '.'         |
+    | '-'         |
+    | '#$'        |
+    | '_VM'       |
+    | ' test'     |
+    | '@test'     |
+    | '!profile'  |
+    | 'demo@1'    |
+    | '?aaa#'     |
+    | '%pro%'     |
+    | '&foo'      |
+    | '*te$st^'   |
+    | 'foo 123'   |
+    | 'foo    '   |
+
+  Scenario Outline: As user, I can create profile with alphanumeric character including '_' and '-' in profile name
+    Given profile "minishift" has state "Does Not Exist"
+     When executing "minishift profile set <profilename>" succeeds
+     Then profile <profilename> should have state "Does Not Exist"
+     And profile <profilename> should be an active profile
+
+  Examples: Correct profile names
+    | profilename |
+    | Test-123    |
+    | profile_    |
+    | P1_name-    |
+    | vm_45       |
+    | 20          |
+    | random001z  |
+    | 00_XYZ-     |
+    | test_Pro-01 |
+    | foo         |
+
+  Scenario: As user, I can switch between existing profiles
+    Given profile "Test-123" has state "Does Not Exist"
+     When executing "minishift profile set Test-123" succeeds
+     Then profile "Test-123" should be an active profile
+     And executing "minishift profile set minishift" succeeds
+     Then profile "minishift" should be an active profile
+
+  Scenario: As user, I can start Minishift with default profile 'minishift'
+    Given profile "minishift" has state "Does Not Exist"
+    And profile "minishift" is an active profile
+     When executing "minishift start" succeeds
+     Then profile "minishift" should have state "Running"
+     And executing "minishift profile list" succeeds
+     Then stdout should match "-\s*minishift\s*Running\s*\(Active\)"
+
+  Scenario: As user, I can apply independent settings in profile 'foo'
```

Comment:
So in this case, given your assertions, where is the proof that the settings are independent? You make some changes to the foo profile and assert that they apply. But to show that these settings are independent you need to also assert that the settings for another profile are unchanged. 

So  a proper test would look imo like:

* Assert that the memory of profile minishift is <n>GB
* Start a new profile 
* Set a new memory size for the new profle
* Assert that the new profile has the specified memory setting
* Assert that the minishift profile still has a memory setting of <n> GB

If you skip the whole business around disk-size, cpus, etc you end up with a test which is about the same length, but actually verifies what you want to show. There is not really much value in setting 4 different config values. What you want to show can be shown with a single one.